docs/proposals/2170-kubeflow-training-v2/README.md

```diff
@@ -301,29 +301,42 @@ type TrainJobSpec struct {
 	Suspend *bool `json:"suspend,omitempty"`
 
 	// ManagedBy is used to indicate the controller or entity that manages a TrainJob.
+	// The value must be either an empty, 'training-operator.kubeflow.org/trainjob-controller' or
+	// 'kueue.x-k8s.io/multikueue'.
+	// The built-in TrainJob controller reconciles TrainJob which don't have this
+	// field at all or the field value is the reserved string
+	// 'training-operator.kubeflow.org/trainjob-controller', but delegates reconciling TrainJobs
```

Comment:
`training-operator.kubeflow.org/trainjob-controller` is a bit long. What about the alternative: `kubeflow.org/training-operator-v2`? 

So, to match the naming of `v1` we tentatively go with for v1: `kubeflow.org/training-operator` (https://github.com/kubeflow/training-operator/issues/2193)?

Reply:
Maybe we should just do `kubeflow.org/training-operator` for V2?
Since we are planning to support only `managedBy: kueue.x-k8s.io/multikueue` in V1 version: https://github.com/kubeflow/training-operator/issues/2193#issuecomment-2273478051

Reply:
Actually, we could probably use `kubeflow.org/training-operator` both for v1 and v2. It is non-ambigues since API v1 can only be reconciled by the controller in the same version.

Reply:
TBH, I'm ok with either name `kubeflow.org/training-operator` or `training-operator.kubeflow.org/trainjob-controller` because I just wanted to align with the managedBy specifications in the batch/job and jobset.

In the batch/job and jobset, we use the `${OPERATOR_NAME}.${API_GROUP}/${CONTROLLER_NAME}` like [`jobset.sigs.k8s.io/jobset-controller`](https://github.com/kubernetes-sigs/jobset/blob/8ab973806b2e7950fcd6f4bdb3731866265425ee/api/jobset/v1alpha2/jobset_types.go#L111-L123) and [`kubernetes.io/job-controller`](https://kubernetes.io/docs/concepts/workloads/controllers/job/#delegation-of-managing-a-job-object-to-external-controller).

@mimowo Do you know the reason why we selected the above managedBy format in the batch/job and jobset?

Reply:
So, let's use the `kubeflow.org/training-operator` as a reserved name so that we can align with batch/job and jobset.

Reply:
Done.

Reply:
Yeah, maybe we should use the `trainjob-controller` to make it consistent with the CRD name, JobSet is doing the same: https://github.com/kubernetes-sigs/jobset/blob/main/api/jobset/v1alpha2/jobset_types.go#L46